influxdb3_write/src/write_buffer/buffer_segment.rs

```diff
@@ -201,15 +342,23 @@ pub(crate) fn load_buffer_from_segment(
                         );
 
                     let db_name = &write.db_name;
-                    if !buffered_data.database_buffers.contains_key(db_name) {
-                        buffered_data.database_buffers.insert(
+                    if !loaded_buffer
+                        .buffered_data
+                        .database_buffers
+                        .contains_key(db_name)
+                    {
+                        loaded_buffer.buffered_data.database_buffers.insert(
                             db_name.clone(),
                             DatabaseBuffer {
                                 table_buffers: HashMap::new(),
                             },
                         );
                     }
-                    let db_buffer = buffered_data.database_buffers.get_mut(db_name).unwrap();
+                    let db_buffer = loaded_buffer
+                        .buffered_data
+                        .database_buffers
+                        .get_mut(db_name)
+                        .unwrap();
```

Comment:
That would actually be really nice. We ran into this issues a couple of times and `entry_ref` would be great. Since std's hashmap is just hashbrown under the hood we won't lose out.

Reply:
yeah, that's why I'm doing it. Moving to using hashbrown sounds good.